Chore: Upgrade and refactor pool for v3.1.0

confleux opened the pull request:
## Summary

This PR modernizes ADAMANT Forging Pool for the `3.1.0` release and refreshes the existing upgrade/refactor branch against the current `dev` baseline.

Closes #5.

## Changes

- Update root, server, web, and migration package metadata for `3.1.0`, including descriptions, keywords, repository links, author metadata, engine requirements, and current dependency ranges
- Refresh dependencies with `npm-check-updates`, install with lifecycle scripts disabled, and remove known npm audit vulnerabilities
- Replace legacy ESLint config files with flat ESLint configs, add Prettier config, and update formatting/lint scripts
- Remove WindiCSS from the web app, add a Svelte config, preserve the existing dashboard utility styles manually, and keep Vite/Svelte builds warning-free
- Restore dashboard reset styles after WindiCSS removal and align numeric voter table headers and values to the right
- Update the Svelte mount target guard so the app fails clearly if `#app` is missing
- Continue the LowDB-to-MongoDB runtime refactor, keep LowDB helpers scoped to migration/legacy tests, and add Mongo-backed repository wiring and indexes
- Update payout, reward distribution, block parsing, cron, notification, config, and store helpers for clearer behavior, safer error handling, and current module structure
- Add and update focused tests for Mongo-backed reward/block flows, cron initialization, delegate helper formatting, and store behavior
- Expand and reorganize README, CONTRIBUTING, AGENTS.md, and config comments for current setup, operations, contribution workflow, links, and repo governance
- Clean up user-facing docs, log messages, notification wording, and comments/JSDoc around changed code paths

## Risk Areas

- Reward distribution and payout accounting paths
- MongoDB persistence, indexes, and migration flow from LowDB data
- Runtime config validation and operator setup docs
- Dashboard dependency/tooling changes after removing WindiCSS
- Notification/log wording changes around pool and delegate names

## Validation

- `npm --prefix server run lint`
- `npm --prefix server test`
- `npm --prefix web run lint`
- `npm run build:web`
- `npm audit`
- `npm --prefix server audit`
- `npm --prefix web audit`
- `npm --prefix scripts/migrate-lowdb-mongodb audit`
- `npm run format:check`
- `git diff --check`
- `npx --yes npm-check-updates --packageFile package.json`
- `npx --yes npm-check-updates --packageFile server/package.json`
- `npx --yes npm-check-updates --packageFile web/package.json`
- `npx --yes npm-check-updates --packageFile scripts/migrate-lowdb-mongodb/package.json`

All audits reported `0 vulnerabilities`, and the final `npm-check-updates` pass reported all package roots up to date.

Jest still prints the existing `--forceExit` notice from the package script after the test suite passes.

## Notes

- Install and lockfile refreshes were run with lifecycle scripts disabled via `--ignore-scripts`
- No dependency lifecycle scripts were manually allowed or rebuilt
- The latest patch dependency refresh updates `prettier` to `^3.9.4` and `eslint-plugin-jest` to `^29.15.4`
- A delegate rank inconsistency observed from `delegates/get` is intentionally not worked around in this repository; it should be fixed in ADAMANT Node so the pool displays the API truth directly

confleux committed:
chore: add dev files and dirs

confleux committed:
chore: update dev deps

confleux committed:
chore: update eslint 8 -> 9

confleux committed:
chore: init husky again to remove deprecated stuff

confleux committed:
chore: add tests script

confleux committed:
chore: minor update of dep

confleux committed:
chore: update vite v4 -> v7

confleux committed:
feat: add link to explorer at address cell

confleux committed:
chore: add links at transation table

confleux committed:
chore: mark whole number part bold

confleux committed:
chore: mark whole number part bold

confleux committed:
chore: update enpoints naming

confleux committed:
chore: update svelte 3 -> 5

confleux committed:
chore: update adamant-api

confleux committed:
fix: refresh button

confleux committed:
chore: minor deps update

confleux committed:
chore: update small deps

confleux committed:
chore: update eslint to v9

confleux committed:
chore: update express

confleux committed:
chore: update lowdb

confleux committed:
chore: add tags to README

confleux committed:
chore: update npm scripts and resolve eslint issues

confleux committed:
chore: update husky pre-commit

confleux committed:
fix: setInterval

confleux committed:
fix: setIntreval ms

confleux committed:
chore: add stylistic config

confleux committed:
chore: rename, make exit code constant

confleux committed:
chore: update cors options

confleux committed:
chore: update naming

confleux committed:
refactor: restructure, use Error instead of 0s and -1s

confleux committed:
refactor: remove unused

confleux committed:
fix: import, naming

confleux committed:
refactor: naming

confleux committed:
refactor: make db async

confleux committed:
refactor: lowdb -> mongodb

confleux committed:
refactor: remove unused LowDB queries

confleux committed:
refactor: move lowb related code to repository dir

confleux committed:
fix: import

confleux committed:
chore: update mongo queries error handling

confleux committed:
chore: make mongodb params configurable

confleux committed:
chore: add comments

confleux committed:
fix: blocks checking

confleux committed:
chore: add migration guide

confleux committed:
chore: add indexes

adamantmm commented:
Let's deploy for the dignity_pool http://138.201.152.191:36667/ and test.
Once we confirm that everything works fine, we'll merge and then add smaller tasks for further changes.

adamantmm commented:
<img width="2502" height="1344" alt="image" src="https://github.com/user-attachments/assets/6d2654b4-6889-4d08-9d61-6d5cb02291f6" />

Compare with:
http://145.223.23.46:36667/

<img width="1198" height="564" alt="image" src="https://github.com/user-attachments/assets/8104e522-5ce4-4007-b1f0-f86fb4a707a0" />

And same for "Transactions" table.

@confleux

adamantmm commented:
It seems payouts not working.
Latest payout 2025-08-01.

<img width="1221" height="377" alt="image" src="https://github.com/user-attachments/assets/0b5a7a79-b101-4ebe-865a-f32b4deec141" />

adamantmm reviewed (changes requested):
See comments above.

adamantmm commented:
Let's also set whole part bold (**12**.3456) in the header values.

<img width="1189" height="303" alt="image" src="https://github.com/user-attachments/assets/0de06051-fb23-4101-bb4c-9bb44243f7d7" />

confleux committed:
fix: payouts

massivedev0 committed:
Merge remote-tracking branch 'origin/dev' into chore/small-updates

# Conflicts:
#	.gitignore

massivedev0 committed:
chore: update package.json for version 3.1.0 and dependency upgrades

- Updated project name and description in package.json.
- Bumped version to 3.1.0.
- Updated Node and npm engine requirements.
- Upgraded various dependencies including ESLint, Svelte, and SMUI components.
- Changed linting commands to use the latest ESLint syntax.

fix: improve DashboardItem styling

- Refactored class binding for text size in DashboardItem.svelte.
- Removed unnecessary style definitions for text sizes.

massivedev0 committed:
chore: clean up package-lock.json and package.json, remove unused dependencies and windi.config.js

massivedev0 committed:
chore: refactor main.js to improve target element handling

massivedev0 committed:
chore: add formatDelegateName and normalizeDelegateRank functions with tests

massivedev0 committed:
chore: remove deprecated node URL from configuration

massivedev0 committed:
chore: update references from ADAMANT Foundation to ADAMANT developer community in configuration and UI

massivedev0 committed:
chore: enhance notification logging for Slack message failures

massivedev0 committed:
chore: update CSS styles for consistency and improve layout handling

massivedev0 committed:
chore: improve logging message format, enhance delegate rank resolution, and update VoterTable for numeric alignment

massivedev0 committed:
chore: enhance numeric header cell alignment in data table

massivedev0 committed:
chore: remove unused delegate rank resolution logic and related tests

massivedev0 committed:
chore: remove unused node URL from configuration

massivedev0 committed:
chore: refresh latest patch dependencies

Update root Prettier and server eslint-plugin-jest patch versions after the final npm-check-updates stop check.

dev-adamant-im reviewed (commented):
Automated code review of v3.1.0 upgrade/refactor. Overall solid: the LowDB→MongoDB migration is consistent across modules, the `adamant-api` v3 response-shape changes (`.data.x` → `.x`, `payment.transactionId`) match the installed library, and `config/validate.js` fixes a real latent bug (the old `forEach` made `return` a no-op so validation never reported errors). Two issues below worth addressing before merge.

dev-adamant-im commented on server/src/api/index.js:
```diff
@@ -1,23 +1,24 @@
-app.use(cors());
+app.use(cors({
+  origin: config.origin,
+  credentials: config.credentials,
```
**CORS config is read from the wrong path.** The schema (`config/schema.js`) and `config.default.jsonc` define these under a `cors` object:
```jsonc
"cors": { "origin": "*", "credentials": true }
```
but here you read `config.origin` / `config.credentials` (top-level), which are `undefined`. The `cors` package treats `origin: undefined` as "reflect/allow all", so it *appears* to work with the default — but any operator who restricts `cors.origin` to a specific domain (or sets `credentials`) is silently ignored and all origins are still allowed. Should be:
```js
app.use(cors({
  origin: config.cors.origin,
  credentials: config.cors.credentials,
  methods: ['GET'],
}));
```

dev-adamant-im commented on scripts/migrate-lowdb-mongodb/index.js:
```diff
@@ -0,0 +1,41 @@
+    const db = new Low(new JSONFile(`${lowdbStoragePath}/${collectionName}.json`), {});
+    await db.read();
+
+    const collection = mongoClient.db('adamant-pool').collection(collectionName);
```
Database name is hardcoded to `'adamant-pool'`, so this ignores the configurable `mongodb.dbName` / connection-string database. If an operator runs the server against a non-default db name, the migration writes to the wrong database and the server sees no data. Consider deriving the db name from an env var (e.g. `MONGODB_DB`) or `client.db()` of the URI default, consistent with the server's `config.mongodb.dbName`.

Two smaller things in this script: `insertMany` on line 36 will throw on the unique `blocks.id` / `voters.address` indexes if the script is re-run (no upsert/`ordered:false`), and there's no guard for unset `MONGODB_URI`/`LOWDB_STORAGE_PATH` (line 17) — `new MongoClient(undefined)` fails with an opaque error.

massivedev0 replied:
Fixed locally.

The LowDB to MongoDB migration now requires explicit environment input, supports `MONGODB_DB` or a database name embedded in `MONGODB_URI`, uses that database instead of the hardcoded `adamant-pool`, creates the unique indexes needed for migrated collections, and writes via `bulkWrite`/`replaceOne` with `upsert` so reruns do not fail on existing keyed records. The README migration example now includes `MONGODB_DB=adamant-pool`.

Validated with:

```sh
node --check scripts/migrate-lowdb-mongodb/index.js
npm --prefix server run lint
npm --prefix server test
git diff --check
```

massivedev0 committed:
chore: update MongoDB migration script and add tests; fix CORS configuration
